### ISSUE TITLE
Prepare Analytical Dataset

### ISSUE THREAD

**Issue by Kgable** [Kgable/Soc-413-Research-Project#6]
Dear @AaronGullickson, 
Thank you for your help. This assignment is done and all work has been committed and pushed to git and is ready for review. 

Thank you,
Kevin Gable

**Comment by AaronGullickson** [comment]
Looks good but don't you need age to get birth cohort?

**Comment by Kgable** [comment]
Good morning @AaronGullickson 
I guess I will grab that too. I assume I need to go to GSS and grab that. When I downloaded and unzipped my extract it was empty. I can't remember how we fixed that last time. 
Are we having class today?

Thank you,
Kevin Gable

**Comment by Kgable** [comment]
I tried to download the age data in a new extract. When I unzipped it, I got an empty file. Is it possible to do a Zoom meeting this week for about 5 minutes? I ran into this before and I can't remember what I did wrong.

Thank you,
Kevin Gable

**Comment by Kgable** [comment]
Good morning @AaronGullickson,
  I hope all is well. I do appreciate your help. I am still trying to unzip my tar.gz file on my Mac. I completed another extract with age included. I can't believe I forgot it. When I double click it to unzip, the folder is empty. I am not sure how to fix it. I have looked on line and haven't found useful suggestions yet. My plan is to unzip this file, replace the data in my raw data folder, then code age in groups on ten years, 0-18 = "under 18", 18-28 = "18 -28 year-olds", "...", 78 or above = "Over 78". Does this makes sense to do? I am thinking about the analysis stage and want to be sure I look at birth cohorts correctly. I will keep the age as a numeric as well just in case. 
   I just need to unzip this file. It is the same issue I had last time. I know the data is there, 194Mb of data is shown to be there. Do you remember how we fixed this last time? 

Thank you,
Kevin Gable

**Comment by AaronGullickson** [comment]
I would code based on birth cohort rather than age in ten year increments. You can derive birth cohort by taking year-age and then you can use the cut command in R to create 10-year slices, something like:

```r
gss$bcohort <- gss$year-gss$age
gss$bcohort10 <- cut(gss$bcohort, seq(from=1900, to=2010, by=10), include.lowest=TRUE)
```
Its possible you might need to collapse the earliest birth cohorts into a larger group if they are very small, which you can do by adjusting that second argument a bit.

On downloading the data from GSS - I am guessing its just a problem with the horrible GSS Explorer system and you are not DLing the right thing or setting up the extract right. You have to be sure to click on the extract you want in the extract tab and then use Action > Download.

**Comment by Kgable** [comment]
Good afternoon @AaronGullickson 

   I hope all is well. I was able to get the file downloaded and unzipped. You were correct! I reloaded my data into my raw folder and read in the data. So far, all my other variables are working fine. When looking at age, I read it in and coded for missing variables. When I run the table command to check, table(gss$AGE, gss$age, exclude=NULL), I get a strange response. I get, [ reached getOption("max.print") -- omitted 62 rows ]. My table result stops giving data after age 27. This surprised me because the missing variables are coded the same on GSS, anything less than zero is NA. Am I doing something wrong? It works for the other variables.
   I put in the code for age cohorts. I am considering how to adjust the second argument. My lowest values are at the age of 18 and ages 78 and older. I still need to run this code so I can see what I might need to collapse. I am predicting older cohorts. That will be fun. I am so close.

Thank you,
Kevin Gable
